src/module.ts

```diff
@@ -16,6 +17,23 @@ export default defineNuxtModule<AlgoliaOptions>({
       throw new Error('Missing `applicationId`')
     }
 
+    if (options.crawler) {
+      if (!options.crawler.apiAdminKey) {
+        throw new Error('Missing `crawler.apiAdminKey`')
+      }
+
+      if (!options.crawler.indexName) {
+        throw new Error('Missing `crawler.indexName`')
+      }
+
+      const pages: CrawlerPage[] = []
+
+      nuxt.addHooks({
```

Comment:
Also, you could create a new page here called 4.advanced.md with the explanation of how crawler works and how the user can utilize it to extend their needs -> https://github.com/nuxt-modules/algolia/tree/main/docs/pages